tasmota/xsns_73_hp303b.ino

```diff
@@ -29,104 +29,113 @@
 #define XSNS_73 73
 #define XI2C_52 52 // See I2CDEVICES.md
 
-#define HP303B_ADDR1 0x77
-#define HP303B_ADDR2 0x76
-
 #include <LOLIN_HP303B.h>
-// HP303B Opject
+// HP303B Object
 LOLIN_HP303B HP303BSensor = LOLIN_HP303B();
-uint8_t bhp303b_addresses[2] = {HP303B_ADDR1, HP303B_ADDR2};
+
+#define HP303B_MAX_SENSORS    2
+#define HP303B_START_ADDRESS  0x76
+
+struct {
+char types[7] = "HP303B";
+uint8_t count  = 0;
+int16_t oversampling = 7;
+} hp303b_cfg;
 
 struct BHP303B {
   uint8_t address;
-  uint8_t type = 0;
   uint8_t valid = 0;
-  float temperature;
-  float pressure;
-  int16_t oversampling = 7;
-  char name[7] = "HP303B";
-} bhp303b_sensor;
+  float temperature = NAN;
+  float pressure = NAN;
+} hp303b_sensor[HP303B_MAX_SENSORS];
 /*********************************************************************************************/
 
-bool HP303B_Read()
+bool HP303B_Read(uint8_t hp303b_idx)
 {
-  if (bhp303b_sensor.valid) { bhp303b_sensor.valid--; }
+  if (hp303b_sensor[hp303b_idx].valid) { hp303b_sensor[hp303b_idx].valid--; }
 
   float t;
   float p;
   int16_t ret;
 
-  ret = HP303BSensor.measureTempOnce(t, bhp303b_sensor.oversampling);
+  ret = HP303BSensor.measureTempOnce(t, hp303b_cfg.oversampling);
   if (ret != 0)
     return false;
 
-  ret = HP303BSensor.measurePressureOnce(p, bhp303b_sensor.oversampling);
+  ret = HP303BSensor.measurePressureOnce(p, hp303b_cfg.oversampling);
   if (ret != 0)
     return false;
 
-  bhp303b_sensor.temperature = (float)ConvertTemp(t);
-  bhp303b_sensor.pressure = (float)ConvertPressure(p) / 100; //conversion to hPa
+  hp303b_sensor[hp303b_idx].temperature = (float)ConvertTemp(t);
+  hp303b_sensor[hp303b_idx].pressure = (float)ConvertPressure(p) / 100; //conversion to hPa
 
-  bhp303b_sensor.valid = SENSOR_MAX_MISS;
+  hp303b_sensor[hp303b_idx].valid = SENSOR_MAX_MISS;
   return true;
 }
 
 /********************************************************************************************/
 
 void HP303B_Detect(void)
 {
-  for (uint32_t i = 0; i < sizeof(bhp303b_addresses); i++)
+  for (uint32_t i = 0; i < HP303B_MAX_SENSORS; i++)
   {
-    if (I2cActive(bhp303b_addresses[i])) { return; }
-
-    bhp303b_sensor.address = bhp303b_addresses[i];
+    if (I2cActive(HP303B_START_ADDRESS + i)) { return; }
 
```

Comment:
us continue to stay in the loop and use `I2cSetDevice`:
`if (!I2cSetDevice(HP303B_START_ADDRESS + i )) { continue; }`

Reply:
Fixed it.

It was unclear to me which one of the two to use as I found both implementations in different sensors.